integration/benchnet2/automate/bootstrap_automation.go

```diff
@@ -0,0 +1,230 @@
+var TEMPLATE_PATH string = "templates/"
+
+var DEFAULT_ACCESS_IMAGE string = "gcr.io/flow-container-registry/access:v0.27.6"
+var DEFAULT_COLLECTION_IMAGE string = "gcr.io/flow-container-registry/collection:v0.27.6"
+var DEFAULT_CONSENSUS_IMAGE string = "gcr.io/flow-container-registry/consensus:v0.27.6"
+var DEFAULT_EXECUTION_IMAGE string = "gcr.io/flow-container-registry/execution:v0.27.6"
+var DEFAULT_VERIFICATION_IMAGE string = "gcr.io/flow-container-registry/verification:v0.27.6"
+
+// Generates a values file based on a given json and templates
+func GenerateValuesYaml(inputJsonFilePath string, templatePath string, outputYamlFilePath string, branch string, commit string) {
+	GenerateValuesYamlWithImages(inputJsonFilePath, templatePath, outputYamlFilePath, branch, commit, DEFAULT_ACCESS_IMAGE, DEFAULT_COLLECTION_IMAGE, DEFAULT_CONSENSUS_IMAGE, DEFAULT_EXECUTION_IMAGE, DEFAULT_VERIFICATION_IMAGE)
+}
+
+// Generates a values file based on a given json and templates, with addtion of selecting image version for nodes
+func GenerateValuesYamlWithImages(inputJsonFilePath string, templatePath string, outputYamlFilePath string, branch string, commit string,
+	accessImage string, collectionImage string, consensusImage string, executionImage string, verificationImage string) error {
+	templateFolder := TEMPLATE_PATH
+	if templatePath != "" {
+		templateFolder = templatePath
+	}
+	nodeInfoJson := DEFAULT_NODE_INFO_PATH
+	if inputJsonFilePath != "" {
+		nodeInfoJson = inputJsonFilePath
+	}
+	nodesData, nodeConfig, err := loadNodeJsonData(nodeInfoJson)
+	if err != nil {
+		return err
+	}
+
+	values, err := buildValuesStruct(templateFolder, nodesData, nodeConfig, branch, commit, accessImage, collectionImage, consensusImage, executionImage, verificationImage)
+	if err != nil {
+		return err
+	}
+
+	return marshalToYaml(values, outputYamlFilePath)
+}
+
+// Loads node info json file and returns unmarshaled struct and node counts
+func loadNodeJsonData(nodeInfoPath string) (map[string]Node, map[string]int, error) {
+	jsonFileBytes, err := textReader(nodeInfoPath)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var nodes []Node
+	err = json.Unmarshal(jsonFileBytes, &nodes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var nodeConfig = make(map[string]int)
+	nodeConfig["access"] = 0
+	nodeConfig["collection"] = 0
+	nodeConfig["consensus"] = 0
+	nodeConfig["execution"] = 0
+	nodeConfig["verification"] = 0
+
+	nodeMap := map[string]Node{}
+	nodeNameRe := regexp.MustCompile(`\w{6,}\d{1,3}`)
+	nodeTypeRe := regexp.MustCompile(`\D{6,}`)
+	for _, node := range nodes {
+		name := nodeNameRe.FindStringSubmatch(node.Address)
+		nodeType := nodeTypeRe.FindStringSubmatch(node.Address)
+		nodeMap[name[0]] = node
+		nodeConfig[nodeType[0]]++
+	}
+
+	return nodeMap, nodeConfig, err
+}
+
+// Builds and returns values yaml struct
+func buildValuesStruct(templateFolder string, nodesData map[string]Node, nodeConfig map[string]int, branch string, commit string,
```

Comment:
does this assume there will a single node of each type? What if I wanted a different number of access and verification nodes?